### DISCUSSION TITLE
Defining a list of roles for Divina

### DISCUSSION THREAD

**Discussion by HadrienGardeur** [readium/guided-navigation#1] [General]
[The initial list of roles for this project](https://github.com/readium/guided-navigation/blob/main/schema/roles.schema.json) is directly based on [DPUB ARIA 1.0](https://www.w3.org/TR/dpub-aria-1.0/), which doesn't really cover anything comic-specific.

Some roles remain useful (`cover` or `pagebreak` for example) but there's nothing really meant to convey the structure within a page.

Looking at the [EPUB 3 Structual Semantics Vocabulary 1.1](https://www.w3.org/TR/epub-ssv-11), there's a short [list of roles dedicated to comics](https://www.w3.org/TR/epub-ssv-11/#sec-comics) inherited from the work on [EPUB Region-Based Navigation 1.0](https://idpf.org/epub/renditions/region-nav):

<dl>
<dt>balloon</dt>
<dd>An area in a comic panel that contains the words, spoken or thought, of a character.</dd>
<dt> panel</dt>
<dd>An individual frame, or drawing.</dd>
<dt> panel-group</dt>
<dd>A group of panels (e.g., a strip).</dd>
<dt> sound-area</dt>
<dd>An area of text in a comic panel that represents a sound.</dd>
<dt> text-area</dt>
<dd>An area of text in a comic panel. Used to represent titles, narrative text, character dialogue (inside a balloon or not) and similar.</dd>
</dl>

Here's my take on this:

- the concept of page and spread are missing and even though spread can be inferred from the presence of `spread` we could also have comics where a single image is used for a spread
- `panel` and `panel-group` are fine as structural elements but we might need something even more generic than panel, since a number of comics do not use panels at all
- `balloon`, `sound-area` and `text-area` feel confusing. `text-area` is probably a superset of `bubble` (I prefer that term to `balloon`) but we need to further qualify different types of bubbles and potential other text areas

I expect this to be a long and iterative process, in which we should probably try to identify the higer-level structural elements first before we dive any deeper.

**Comment by HadrienGardeur** [comment]
Here's a first list of suggested roles.

### Structure

This covers the basic structural building block of a Divina publication.

| Role | EPUB type equivalent | 
| ---- | ---------------------- | 
| `spread` | |
| `page` | | 
| `panelGroup` | [`panel-group`](https://www.w3.org/TR/epub-ssv-11/#sec-comics) | 
| `panel` | [`panel`](https://www.w3.org/TR/epub-ssv-11/#sec-comics) | 

I think it's worth pointing out that `img` in ARIA is quite similar `spread`, since it's meant to be used over multiple images that should be presented together:

```html
<div role="img" aria-label="Description of the overall image">
  <img src="graphic1.png" alt="" />
  <img src="graphic2.png" />
</div>
```

### Text

| Role | EPUB type equivalent | 
| ---- | ---------------------- | 
| `bubble` | [`balloon`](https://www.w3.org/TR/epub-ssv-11/#sec-comics) | 
| `caption` | | 
| `sound` | [`sound-area`](https://www.w3.org/TR/epub-ssv-11/#sec-comics) | 
| `shoutBubble` | | 
| `speechBubble` | | 
| `thoughtBubble` | | 

`bubble` is meant to be used for bubbles/balloons who do not have a specific role (`speechBubble`, `thoughtBubble` or `shoutBubble`) to identify them.

`caption` is meant to either identify a caption below a panel or within a panel (usually a narrator). It would be inherited from `<caption>` and `<figcaption>` in HTML.

**Reply by llemeurfr** [reply]
I'm not a fan of "caption" for the information given by a narrator (I have Alix in mind here, e.g. [this page](https://www.bdnet.com/img/couvpage/12/9782203312128_pg.jpg)). 
In French it is a "récitatif" (see [this lexicon](https://dtna.canoprof.fr/eleve/Arts%20et%20Culture/PREAC%20BANDE-DESSINEE/res/Lexique2.pdf)).
In english "narration" could fit.

**Reply by llemeurfr** [reply]
Apart from that I'm ok with this vocabulary (as a start).

**Reply by HadrienGardeur** [reply]
> I'm not a fan of "caption" for the information given by a narrator

Based on what I've been able to identify so far, "caption" seems to be the correct term in English. This page does a great job of providing a full vocabulary for comics: https://blambot.com/pages/comic-book-grammar-tradition

**Reply by llemeurfr** [reply]
ok then; I had not find a proper English lexicon.

**Reply by crigaud** [reply]
Regarding text, shoutBubble, speechBubble and thoughtBubble being at the same role level than bubble is strange for me. Would it be possible to qualify bubble with shout, speech and thought instead?

Also, I feel comic characters are missing, along with their link to bubble and text.

**Reply by HadrienGardeur** [reply]
> Regarding text, shoutBubble, speechBubble and thoughtBubble being at the same role level than bubble is strange for me. Would it be possible to qualify bubble with shout, speech and thought instead?

Since roles are represented as an array of strings, we can't represent any kind of structure.

In this proposal `bubble` is the generic (and fallback) role for bubbles, while `shoutBubble`, `speechBubble` and `thoughtBubble` are meant to be specific (we could extend the list if you think that's necessary).

> Also, I feel comic characters are missing, along with their link to bubble and text.

This would be handled through a dedicated `character` property which I have yet to submit a proposal for. At a bare minimum, this should also allow an array of strings (the names of these characters), but I'd like to explore an object representation as well which could contain: `name`, `identifier`, `gender` and `age`.

Take a look at the following comment if you're looking for an example: https://github.com/readium/guided-navigation/issues/8#issuecomment-2239066761

**Comment by HadrienGardeur** [comment]
I've created a complete example at: https://github.com/readium/guided-navigation/blob/main/examples/Pepper%20%26%20Carrot/guided.json

In terms of semantics, this was a very straightforward example where only `panel`, `speechBubble` and `shoutBubble` were needed.
